src/ClientHandle.cpp

```diff
@@ -1863,6 +1863,12 @@ void cClientHandle::RemoveFromWorld(void)
 	m_LastStreamedChunkZ = 0x7fffffff;
 
 	m_HasSentPlayerChunk = false;
+
+	// Do not send chunks for 1.5 seconds.
+	// Apparently, the chunks of the old world fail to unload if
+	// chunks in the new world which have the same coords are sent too fast after the unload.
+	// this is a temp fix, find the root cause!
+	m_ChunkSendDelay = 30;
```

Comment:
Well, we aren't sending a respawn packet at all in same dimension multiworld, so that might have something to do with it. What we do instead is send unload chunk packets, then reload chunk packets. Clearly the client is not processing them in a queue like we expect.